c++11/test/in_memory_recorder.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <mutex>
+#include <stdexcept>
+#include <vector>
+#include "../src/recorder.h"
+
+namespace lightstep {
+// InMemoryRecorder is used for testing only.
+class InMemoryRecorder : public Recorder {
+ public:
+  void RecordSpan(collector::Span&& span) noexcept override {
+    std::lock_guard<std::mutex> lock_guard{mutex_};
+    spans_.emplace_back(std::move(span));
+  }
+
+  std::vector<collector::Span> spans() const {
+    std::lock_guard<std::mutex> lock_guard{mutex_};
+    return spans_;
+  }
+
+  size_t size() const {
```

Comment:
I agree. The only thread-safe way I know is this:
````
template <typename F>
void size(const F& fInLock) const {
    std::lock_guard<std::mutex> lock_guard{mutex_};
    fInLock(spans_.size());
}
````
`foo.size([&](const auto& size) { mySize = size; });`, but it's quite more complex.